client/state/themes/actions.js

```diff
@@ -106,13 +106,53 @@ export function receiveTheme( theme, siteId ) {
  * @return {Object}        Action object
  */
 export function receiveThemes( themes, siteId ) {
-	return {
-		type: THEMES_RECEIVE,
-		themes,
-		siteId
+	return ( dispatch, getState ) => {
+		const { filteredThemes } = filterThemes( getState, themes, siteId );
+		dispatch( {
+			type: THEMES_RECEIVE,
+			themes: filteredThemes,
+			siteId
+		} );
+	};
+}
```

Comment:
Revisiting this, seems like I was wrong about this:

> I don't think we have to filter inside `receiveThemes`, since the reducer will only add themes to the corresponding  `items`, but not `queries` subtree